proposals/NNNN-Processing-of-unknown-enum-values-by-SDL-and-Mobile-libraries.md

```diff
@@ -0,0 +1,176 @@
+      * minimal SDL version
+      * application HMI type(s) (`AppHMIType` array)
+  #### Changes related to sending request from Mobile application to SDL
+  After the connection is established and Mobile Application is registered with SDL core, Mobile Application works as usual but requires the request parameters validation. An issue will occur when the Mobile Application calls a method and passes an enum values array as a parameter that belongs to the higher SDL version. In this case, the Mobile Proxy should filter out unsupported params before sending the request and provide warning in logs as `unsupported_values`.
+  The validation sequence:
+  1. Mobile Application sends request to SDL core through the proxy.
+  2. The proxy checks the request parameters:
+     * If there are no enum values unknown to SDL, the proxy forwards the request to SDL core without changes.
+     
+     * If some of the enum values are unknown to SDL core the proxy should:
+          * filter out all RPC’s unsupported enum values including nested RPC’s if those are present, before making a request to make sure it is compliant with the current spec
+          * log on warning level with «unsupported_values» (enum’s/struct’s names)
+    
+  **Note**: For `RegisterAppInterface` to filter we should use version of PROTOCOL, but not the MOBILE_API according to system’s flow. **iOS** *should be extended with similar mechanism in order to encapsulate filtering logic.*
+
+  **For example** : 
+  If the UI app requested unsupported `AppHMIType` and version which are not compatible then the proxy must filter unsupported HMI types. 
+
+      - Before filtering :
+          "AppHMIType" : ["DEFAULT", "MEDIA", "UKNOWN"]
+      - After filtering :
+          AppHMIType" : ["DEFAULT", "MEDIA"]
+  ***Note**: In case no HMI types are left after filtering, Mobile library should throw Exception, otherwise RegisterAppInterface RPC will be sent with remaining HMI types.*
+  ***Note**: Now system flow is the following: if `RegisterAppInterface` fails with an error then Mobile library should pass the error status to the Mobile Library adding failure reason*
+  ***Note**: The protocol version control logic partly implemented in Android and not implemented in iOS though it has its flaws and should be updated following the logic above.*
+
+     
+  ## Potential downsides
+  There might be certain corner cases with filtering out RPCs appeared since version greater than current.
+  *Note: For `RegisterAppInterface` we cannot keep connection if invalid param is present due to the current flow of the older SDL versions. The connection can be dropped (End Session) by SDL core but not by Mobile Device (SDL library)*
+      
+  ## Impact on existing code
+  Android/iOS 
+     * Each `RPCStruct` should be extended with the implementation of filtering mechanism
+     * Implement logging for unsupported values
+  ## Alternatives considered
+  N/A
```

Comment:
Each section of the proposal template should be completed.  If you do not have an alternatives, please include "the author was unable to determine any alternative solutions."